Assets/PatchKit Patcher/Scripts/Utilities/GZipReadWrapperStream.cs

```diff
@@ -0,0 +1,101 @@
+using System.IO;
+using System;
+
+namespace PatchKit.Unity.Utilities
+{
+    public class GZipReadWrapperStream : Stream
+    {
+        private readonly int _minimumReadSize;
+
+        private readonly Stream _source;
+
+        public override bool CanRead
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return _source.CanSeek;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override long Length
+        {
+            get
+            {
+                return _source.Length;
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                return _source.Position;
+            }
+            set
+            {
+                _source.Position = value;
+            }
+        }
+
+        public override void Flush()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public GZipReadWrapperStream(Stream sourceStream)
+        {
+            _source = sourceStream;
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            int totalBytesRead = 0;
+            int bytesRead;
+
+            byte[] tempBuffer = new byte[count];
```

Comment:
There is no need for creating additional buffer. You can request from underlaying `_source` to write to `buffer` with certain offset nad smaller `count` to match the missing required bytes.